File: 240719/정수 명령 처리 3/process-numeric-commands-3.py
from collections import deque 
n = int(input())
num = deque() 

for _ in range(n):
    sr = input().split() 
    s = sr[0]
    if sr[0]=='push_front':
        num.appendleft(int(sr[1]))
    elif s == 'push_back':
        num.append(int(sr[1]))
    elif s == 'pop_front':
        print(num.popleft())
    elif s =='pop_back':
        print(num.pop())
    elif s =='size':
        print(len(num))
    elif s =='empty':
        if len(num) == 0:
            print(1)
        else:
            print(0)
    elif s =='front':
        print(num[0])
    else:
        print(num[-1])

File: 241121/정수 명령 처리 3/process-numeric-commands-3.java
import java.util.*;
public class Main {
    public static void main(String[] args) {
        // 여기에 코드를 작성해주세요.
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        Deque<Integer> deque = new ArrayDeque<>();

        for(int i = 0; i<n; i++){
            String str = sc.next();
            if(str.equals("push_front")){
                int k = sc.nextInt();
                deque.addFirst(k);
            }else if(str.equals("push_back")){
                int k = sc.nextInt();
                deque.addLast(k);
            }else if(str.equals("pop_front")){
                System.out.println(deque.pollFirst());
            }else if(str.equals("pop_back")){
                System.out.println(deque.pollLast());
            }else if(str.equals("size")){
                System.out.println(deque.size());
            }else if(str.equals("empty")){
                if(deque.isEmpty()){
                    System.out.println(1);
                }else{
                    System.out.println(0);
                }
            }else if(str.equals("front")){
                System.out.println(deque.peekFirst());
            }else{
                System.out.println(deque.peekLast());
            }

        }

    }
}